carbonmark-api/test/plugins/rate-limit.test.ts

```diff
@@ -14,24 +16,21 @@ describe("Rate Limiter", () => {
   });
 
   test("should limit requests", async () => {
-    for (let i = 0; i < 100 + 1; i++) {
+    for (let i = 0; i < LIMIT + 1; i++) {
       nock(GRAPH_URLS["polygon"].digitalCarbon)
         .post("")
-        .reply(200, { data: { carbonProjects: [] } });
+        .reply(200, { data: { carbonProjects: [] } })
+        .persist(true);
 
-      nock(GRAPH_URLS["polygon"].marketplace)
-        .post("")
-        .reply(200, { data: { categories: CATEGORIES } });
-
-      const response = await app.inject({
-        method: "GET",
-        url: `${DEV_URL}/categories`,
-      });
+      mockMarketplaceArgs();
 
-      if (i < 101) {
-        expect(response.statusCode).toBe(200);
-      } else {
-        expect(response.statusCode).toBe(429);
+      //Because we are throwing errors in mock fetch we need to catch
+      try {
+        const categories = await mock_fetch(app, "/categories");
+        expect(categories.length).toBeGreaterThan(0);
+      } catch (e) {
+        expect(e.message).toMatch("Too Many Requests");
+        expect(e.message).toMatch("Rate limit exceeded, retry in 1 minute");
```

Comment:
Maybe we can also make sure that `1 === 101` to ensure we're not failing earlier than we should.